plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -553,6 +581,76 @@ public ConnectorInsertTableHandle beginInsert(ConnectorSession session, Connecto
                 .collect(toImmutableList())));
     }
 
+    private void cleanExtraOutputFiles(ConnectorSession session, Set<String> writtenFiles)
+    {
+        HdfsContext hdfsContext = new HdfsContext(session);
+        Set<String> locations = getOutputFilesLocations(writtenFiles);
+        for (String location : locations) {
+            cleanExtraOutputFiles(hdfsContext, session.getQueryId(), location, writtenFiles);
+        }
+    }
+
+    private void cleanExtraOutputFiles(HdfsContext hdfsContext, String queryId, String location, Set<String> filesToKeep)
+    {
+        List<String> filesToDelete = new LinkedList<>();
```

Comment:
i don't like a LinkedList, and indeed ArrayDeque could serve the needs here, but OK to keep.
the list doesn't need to be 'the best', as there shouldn't be more than few elements